### PR TITLE
Add more tests to payload/xml

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Includes types that use TypeSpec.Xml decorators.

# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/payload/xml/main.tsp [hunk 1]
```diff
@@ -32,6 +34,42 @@ model ModelWithOptionalField {
   value?: int32;
 }
 
+@doc("Contains fields that are XML attributes.")
+model ModelWithAttributes {
+  @attribute id1: int32;
+  @attribute id2: string;
+  @name("Enabled") enabled: boolean;
```
Comment (timotheeguerin) [inline packages/cadl-ranch-specs/http/payload/xml/main.tsp:41]: i think you have casing discrepancies between the docs on model

Comment (timotheeguerin) [inline packages/cadl-ranch-specs/http/payload/xml/main.tsp:41]: could we keep trhe renaming of props as a separate dedicated test.

Should also add one that is done with `@encodedName("application/xml", "abc")` which should be equivalent to Xml.name

Comment (jhendrixMSFT) [reply packages/cadl-ranch-specs/http/payload/xml/main.tsp]: done

File: packages/cadl-ranch-specs/http/payload/xml/main.tsp [hunk 2]
```diff
@@ -32,6 +34,55 @@ model ModelWithOptionalField {
...
+  counts: int32[];
+}
+
+@doc("Contains fields of wrapped and unwrapped arrays of primitive types that have different XML representations.")
+model ModelWithRenamedArrays {
+  @name("Colors") @unwrapped colors: string[];
+  @name("Counts") counts: int32[];
+}
+
+@doc("Contains fields of the same type that have different XML representation.")
```
Comment (tadelesh) [inline packages/cadl-ranch-specs/http/payload/xml/main.tsp:56]: a little confused about "same type that have different XML representation"

**Review by tadelesh — APPROVED** [review]

File: packages/cadl-ranch-specs/http/payload/xml/main.tsp [hunk 3]
```diff
@@ -130,3 +181,127 @@ interface ModelWithOptionalFieldValue
...
+  extends XmlOperations<
+      ModelWithUnwrappedArray,
+      """
+  <ModelWithUnwrappedArray>
+    <colors>red</colors>
+    <colors>green</colors>
+    <colors>blue</colors>
+    <counts>
+      <int32>1</int32>
+      <int32>2</int32>
```
Comment (tadelesh) [inline packages/cadl-ranch-specs/http/payload/xml/main.tsp:209]: is this the convention to use type name as tag name?

Comment (jhendrixMSFT) [reply packages/cadl-ranch-specs/http/payload/xml/main.tsp:209]: Yes, see https://typespec.io/docs/next/libraries/xml/guide

File: packages/cadl-ranch-specs/http/payload/xml/main.tsp (hunk 2 above)
Comment (jhendrixMSFT) [reply packages/cadl-ranch-specs/http/payload/xml/main.tsp:56]: This type has two fields that are both of type `SimpleModel` but have different XML names, `InputData` and `OutputData`.

It's the same as [this](https://github.com/Azure/azure-rest-api-specs/blob/a58614a53ecd06c6e6638b58c407d5bd8e61c009/specification/storage/data-plane/Microsoft.BlobStorage/stable/2021-12-02/blob.json#L10870-L10880) scenario in storage.

File: packages/cadl-ranch-specs/http/payload/xml/main.tsp (hunk 3 above)
Comment (tadelesh) [reply packages/cadl-ranch-specs/http/payload/xml/main.tsp:209]: i see. thanks for the info.

**Pull request merged**